packages/streamline/src/components/checkbox/checkbox.tsx

```diff
@@ -20,53 +22,83 @@ export const Checkbox = ({
   return (
     <Pressable
       disabled={disabled}
-      style={styles.pressable}
       hitSlop={10}
       onPress={handleOnPress}
       testID={testID}
+      style={styles.pressable}
     >
       {({ pressed }) => (
-        <>
-          {pressed ? (
-            <Box
-              position={'absolute'}
-              top={0}
-              left={0}
-              right={0}
-              bottom={0}
-              backgroundColor={colors.shadowColor}
-              borderRadius={'sm'}
-            />
-          ) : null}
+        <Box
+          alignItems={'center'}
+          justifyContent={'center'}
+          height={24}
+          width={24}
+        >
           <Box
-            height={18}
-            width={18}
-            borderWidth={2}
-            borderRadius={'sm'}
+            position={'absolute'}
+            top={0}
+            left={0}
+            right={0}
+            bottom={0}
+            backgroundColor={colors.shadowColor}
+            style={styles.shadow}
+            opacity={pressed ? 1 : 0}
+          />
+          <Box
+            height={20}
+            width={20}
             alignItems={'center'}
             justifyContent={'center'}
-            borderColor={
-              value || pressed ? colors.activeColor : colors.borderColor
-            }
             backgroundColor={
-              value ? colors.activeColor : colors.backgroundColor
+              value || pressed ? colors.activeColor : colors.borderColor
             }
+            style={styles.border}
           >
-            <Icon iconName={'Check'} color={colors.iconColor} size="small" />
+            <Box
+              height={16}
+              width={16}
+              alignItems={'center'}
+              justifyContent={'center'}
+              backgroundColor={
+                value ? colors.activeColor : colors.backgroundColor
+              }
+              style={styles.container}
+            >
+              <Icon iconName={'Check'} color={colors.iconColor} size="small" />
+            </Box>
           </Box>
-        </>
+        </Box>
       )}
     </Pressable>
   );
 };
 
-const styles = StyleSheet.create({
-  pressable: {
-    width: 22,
-    height: 22,
-    alignItems: 'center',
-    justifyContent: 'center',
-  },
-});
+const useStyles = (): {
+  pressable: ViewStyle;
+  shadow: ViewStyle;
+  border: ViewStyle;
+  container: ViewStyle;
+} => {
+  const { borderRadii } = useStreamlineTheme();
+
+  const shadowRaddi = borderRadii.sm + 2;
+  const containerRaddi = borderRadii.sm - 2;
```

Comment:
Can you add a comment above to explain why we need this? 
You can also refer to the related issues:
https://github.com/facebook/react-native/issues/36566
https://github.com/facebook/react-native/issues/33060

Reply:
This is to have a perfect border radius!

Here is an article about it => https://www.30secondsofcode.org/articles/s/css-nested-border-radius/#:~:text=The%20border%20radius%20of%20the,tersely%20R1%20%2B%20D%20%3D%20R2%20

Reply:
I added constant file to make it more clear

Reply:
Yes! Sorry if it was not clear, can you add a comment in the code to explain why we are doing I like this 
Otherwise all good to me :)

Reply:
My bad, didn't read correctly 🙃